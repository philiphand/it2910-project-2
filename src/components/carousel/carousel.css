.carousel {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row;
    position: relative;
    overflow-y: auto;
}

.waiting-for-media {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;

    font-weight: 300;
    font-size: 48px;
}

.carousel img {
    cursor: pointer;
}

.next-item {
    position: absolute;
    z-index: 8;
    top: calc(50% - 20px);
    right: 20px;
}

.previous-item {
    position: absolute;
    z-index: 8;
    top: calc(50% - 20px);
    left: 20px;
    -webkit-transform:rotate(180deg);
    -moz-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    -o-transform: rotate(180deg);
    transform: rotate(180deg);
}

.view-port {
    overflow: hidden;
}

.items {
    display: flex;
    flex-direction: row;
    width: fit-content;
    transition: transform 800ms;
}

@media only screen and (max-width: 600px) { 
    .waiting-for-media {
        font-size: 28px;
        text-align: center;
    }
}