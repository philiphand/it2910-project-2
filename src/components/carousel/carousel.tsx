import React, { useState } from 'react'
import { InstallationTemplate } from './installations/template/template'
import { templateDraw } from './installations/template/animation'
import { MediaContext } from '../../mediaContext'

export const Carousel:React.FunctionComponent<any> = () => {
    const [runInstallation, setRunInstallation] = useState(false)

    return (<div>
        <MediaContext.Consumer>
            {(mediaHandler) => {
                    if (mediaHandler) {
                        return (<InstallationTemplate 
                            width={400} 
                            height={400} 
                            inputs={{ mediaHandler }} 
                            running={runInstallation} 
                            draw={templateDraw}></InstallationTemplate>
                        )
                    }
                    else {
                        return (<div>Loading...</div>)
                    }
                }
            }
        </MediaContext.Consumer>
        <button onClick={() => setRunInstallation(!runInstallation)}>KJÃ˜R</button>
    </div>)
}