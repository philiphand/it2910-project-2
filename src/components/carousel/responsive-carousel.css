.carousel-responsive-wrapper {
    position: relative;
}