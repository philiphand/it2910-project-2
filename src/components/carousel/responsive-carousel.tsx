import React, { useEffect, useRef, useState } from 'react'
import { Carousel } from './carousel'

import './responsive-carousel.css'

export const ResponsiveCarousel: React.FunctionComponent = () => {
    const [viewportSize, setViewportSize] = useState({ width: 1280, height: 720 })

    const wrapper = useRef<HTMLDivElement>(null)

    useEffect(() => {
        if (wrapper.current !== null) {
          setViewportSize({ width: wrapper.current.clientWidth, height: wrapper.current.clientHeight })
          window.addEventListener("resize", () => {
            if (wrapper.current !== null)
              setViewportSize({ width: wrapper.current.clientWidth, height: wrapper.current.clientHeight })
          })
        }
      }, [setViewportSize])

    return (<div className="carousel-responsive-wrapper" ref={wrapper}>
        <Carousel viewport={viewportSize} />
    </div>)
}