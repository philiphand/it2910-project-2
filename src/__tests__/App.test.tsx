import React from 'react';
import renderer from 'react-test-renderer';
import {Carousel} from '../components/carousel/carousel';

test('renders correctly', () => {
  const component = renderer
    .create(<Carousel></Carousel>);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot();


});