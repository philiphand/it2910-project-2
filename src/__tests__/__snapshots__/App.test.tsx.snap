// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="waiting-for-media"
>
  Press play to see the animation
</div>
`;
